minifi-commons/minifi-commons-schema/src/main/java/org/apache/nifi/minifi/commons/schema/BaseSchema.java

```diff
@@ -15,13 +15,17 @@
  * limitations under the License.
  */
 
-package org.apache.nifi.minifi.bootstrap.util.schema.common;
+package org.apache.nifi.minifi.commons.schema;
 
 import java.lang.reflect.Constructor;
 import java.lang.reflect.InvocationTargetException;
+import java.util.Collections;
+import java.util.LinkedHashMap;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 public class BaseSchema {
```

Comment:
I think I originally should have made this class Abstract. One of the main things is the toMap method. In this class all it does it return a LinkedHashMap and that just feels off to me. I see that others are overriding it and just using this to get a LinkedHashMap. Also I think it is counter intuitive to have the ability to initialize a BaseSchema. 

Overall I believe all the "Value Access/Interpretation helper methods" should be make static and toMap would be abstract. Thoughts?